.github/workflows/release.yml

```diff
@@ -30,15 +54,339 @@ jobs:
+          xcrun notarytool submit "notarization-x64.zip" --keychain-profile "notarytool-profile" --wait
+
+          echo "Attaching staple"
+          xcrun stapler staple "coveo-${{env.cliversion}}-${{env.commitSHA1}}-arm64-signed.pkg"
+          xcrun stapler staple "coveo-${{env.cliversion}}-${{env.commitSHA1}}-x64-signed.pkg"
+
+          echo "Deleting unsigned packages"
+          mv "coveo-${{env.cliversion}}-${{env.commitSHA1}}-x64-signed.pkg" "coveo-${{env.cliversion}}-${{env.commitSHA1}}-x64.pkg"
+          mv "coveo-${{env.cliversion}}-${{env.commitSHA1}}-arm64-signed.pkg" "coveo-${{env.cliversion}}-${{env.commitSHA1}}-arm64.pkg"
+          rm notarization-arm64.zip notarization-x64.zip
+      - name: Upload binaries
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3
+        with:
+          name: release-binaries
+          path: ./packages/cli/core/dist/**/*
+  release-github:
+    needs: [release-git, release-npm, build-binaries]
+    if: ${{needs.release-npm.outputs.shouldReleaseCli}}
+    environment: 'Release'
+    runs-on: ubuntu-20.04
+    env:
+      GITHUB_CREDENTIALS: ${{ secrets.CLI_RELEASE }} # Required to lock master
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3
+        with:
+          # pulls all commits (needed for computing the next version)
+          fetch-depth: 0
+      - uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3
+        with:
+          registry-url: 'https://registry.npmjs.org'
+          cache: 'npm'
+          node-version-file: '.nvmrc'
+      - name: Download NPM artifacts
+        uses: actions/download-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3
+        with:
+          name: npm-release
+      - name: Download binaries artifacts
+        uses: actions/download-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3
+        with:
+          name: release-binaries
+      - name: Apply & delete the patch
+        run: |
+          git apply release.patch
+          rm release.patch
+      - name: Install dependencies
+        run: npm ci
+      - name: Release GitHub
+        run: npm run release:github
+        env:
+          VERSION: ${{ inputs.version }}
+          DEBUG: ${{ inputs.debug && '*' || '' }}
           RELEASER_APP_ID: ${{ secrets.RELEASER_APP_ID }}
           RELEASER_PRIVATE_KEY: ${{ secrets.RELEASER_PRIVATE_KEY }}
           RELEASER_CLIENT_ID: ${{ secrets.RELEASER_CLIENT_ID }}
           RELEASER_CLIENT_SECRET: ${{ secrets.RELEASER_CLIENT_SECRET }}
           RELEASER_INSTALLATION_ID: ${{ secrets.RELEASER_INSTALLATION_ID }}
           DEPLOY_KEY: ${{ secrets.DEPLOY_KEY }}
+      - name: Upload binaries
+        uses: svenstaro/upload-release-action@7319e4733ec7a184d739a6f412c40ffc339b69c7 # 2.5.0
+        with:
+          repo_token: ${{ secrets.GITHUB_TOKEN }}
+          file: ./packages/cli/core/dist/**/*
+          file_glob: true
+          tag: ${{ env.tag }}
```

Comment:
File seems too long. Could it be broken down into smaller logical units or are they all dependent and it makes sense to keep them together?

Reply:
I agree, I do want to refactor it. However, before refactoring it, I think it's best to see the flow work and then cut pieces.
There are more than a few that are dependent on each another too

Reply:
Maybe add a TODO ?